nexus_allowlist/actions.py

```diff
@@ -238,7 +311,19 @@ def recreate_privileges(
             )
             cran_privilege_names.append(privilege_name)
 
-    return pypi_privilege_names + cran_privilege_names
+        # Allow selected APT packages
+        for package in apt_allowlist:
+            privilege_name = create_content_selector_and_privilege(
+                nexus_api,
+                name=f"apt-{package}",
+                description=f"Allow access to {packages} APT package",
+                expression=f'format == "apt" and path=~"^/pool/.*/{package}.*"',
```

Comment:
The regex here feels quite permissive.

What does the directory structure looks like between `/pool/` and `{package`?

Reply:
Here is an exemple :
`https://fr.archive.ubuntu.com/ubuntu/pool/universe/r/r-base/r-base_4.4.2-1_all.deb`
Unless you want to restrict access to certain package archives (main, universe, multiverse, restricted for Ubuntu), I don't see how you can make the regex stricter.

Reply:
I can't remember how flexible the expression language is, but it could be something like `^/pool/(main|universe|multiverse|restricted)/.*/{package}.*"` or even `^/pool/({'|'.join(allowed_archives)})/.*/{package}.*"`.

Reply:
According the [Nexus documentation](https://help.sonatype.com/en/content-selectors.html#content-selector-operators-for-non-ha-deployment), the content selector supports Java Regex. It worths a try.